code/modules/mining/remote_fob.dm

```diff
@@ -0,0 +1,336 @@
+	desc = "A computer console equipped with camera screen and controls for a planetside deployed construction drone. Materials or equipment vouchers can be added simply by inserting them into the computer."
+//	icon_screen = 
+//	icon_keyboard =
+	icon_state = "syndishuttle"
+	req_access = list(ACCESS_MARINE_REMOTEBUILD)
+	networks = FALSE
+	off_action = new/datum/action/innate/camera_off/remote_fob
+	jump_action = null
+	var/drone_creation_allowed = TRUE
+	var/spawn_spot
+	var/lz
+	var/datum/action/innate/remote_fob/metal_cade/metal_cade = new
+	var/max_metal = 50 //mostly to prevent jokers collecting all the metal and dumping it in
+	var/metal_remaining = 0
+	var/datum/action/innate/remote_fob/plast_cade/plast_cade = new
+	var/max_plasteel = 50
+	var/plasteel_remaining = 0
+	var/datum/action/innate/remote_fob/toggle_wiring/toggle_wiring = new //whether or not new barricades will be wired
+	var/do_wiring = FALSE
+	var/datum/action/innate/remote_fob/sentry/sentry = new
+	var/sentry_remaining = 0
+
+/obj/machinery/computer/camera_advanced/remote_fob/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT, .proc/disable_drone_creation)
+
+/obj/machinery/computer/camera_advanced/remote_fob/proc/disable_drone_creation()
+	drone_creation_allowed = FALSE
+	UnregisterSignal(SSdcs, COMSIG_GLOB_DROPSHIP_TRANSIT)
+
+/obj/machinery/computer/camera_advanced/remote_fob/examine(mob/user)
+	. = ..()
+	var/list/details = list()
+	details +=("It has [metal_remaining] sheets of metal remaining.</br>")
+	details +=("It has [plasteel_remaining] sheets of plasteel remaining.</br>")
+	details +=("It has [sentry_remaining] sentries ready for placement.</br>")
+	to_chat(user, "[details.Join(" ")] ")
+
+/obj/machinery/computer/camera_advanced/remote_fob/give_eye_control(mob/user)
+	. = ..()
+	user.lighting_alpha = 120
+	eyeobj.name = "Remote Construction Drone"
+	if(eyeobj.eye_initialized)
+		eyeobj.setLoc(get_turf(spawn_spot))
+
+
+/obj/machinery/computer/camera_advanced/remote_fob/attack_hand(mob/living/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='warning'>Access Denied!</span>")
+		return
+	if(!drone_creation_allowed)
+		to_chat(user, "<span class='notice'>Communication with the drone impossible due to fuel-residue in deployment zone atmosphere.</span>")
+		return
+	spawn_spot = FALSE
+	var/port
+	switch(tgalert(user, "Summon Drone in:", "FOB Construction Drone Control", "LZ1","LZ2"))
+		if("LZ1")
+			port = /obj/docking_port/stationary/marine_dropship/lz1
+			for(port in SSshuttle.stationary)
+				if(istype(get_area(port), /area/shuttle/drop1/lz1))
+					spawn_spot = port
+					break
+		if("LZ2")
+			port = /obj/docking_port/stationary/marine_dropship/lz2
+			for(port in SSshuttle.stationary)
+				if(istype(get_area(port), /area/shuttle/drop2/lz2))
+					spawn_spot = port
+					break
+		else
+			return
+	. = ..()
+	
+
+/obj/machinery/computer/camera_advanced/remote_fob/CreateEye()
+	if(spawn_spot)
```

Comment:
You should handle the case there isnt a spawn spot and offer feedback to the user

Reply:
the case of no spawn spot being there is when attack_hand is called and the resulting popup window is closed rather than a selection being made. offering feedback for that seems useless.
the if check here is to prevent issues i've encountered during testing resulting in createeye being called to the last used spawn_spot, ie you summon a drone to lz1, cancel out, click on the computer, close the window instead of making a selection and instead it gives you a drone at lz1.

Reply:
From what I gather if no selection is made then `attack_hand()` should return, not call parent, in current code.
If it's calling parent something is wrong.

From what you describe, you should consider nulling the `spawn_spot` at some point.

Prefer to use early returns, and to give feedback for behavior that is not supposed to happen. Silent errors are the bane of debugging.
In this case it would go:
```
if(!spawn_spot)
	CRASH("CreateEye() called without a spawn_spot designated")
```
That gives you a nice runtime and the call stack to retrace what went wrong.

Reply:
spawn_spot is nulled every time before a selection is made precisely to prevent this kind of behavior, i encountered it during testing a bunch of times, aswell any other unforeseen behavior

Reply:
Alright. Then just do the early return here.